File: RepoTransBench/source_projects/Python/python-babel_flask-babel/public_tests/test_public_multiple_apps.py
import flask
import flask_babel as babel

def test_public_multiple_apps():
    b = babel.Babel()

    app1 = flask.Flask(__name__)
    b.init_app(app1, default_locale="fr_FR")

    app2 = flask.Flask(__name__)
    b.init_app(app2, default_locale="it_IT")

    with app1.test_request_context():
        assert str(babel.get_locale()) == "fr_FR"
        # Data changed: Use a different message and name; French translation should be fallback to original since no translation
        assert babel.gettext("Welcome %(user)s!", user="Marie") == "Welcome Marie!"

    with app2.test_request_context():
        assert str(babel.get_locale()) == "it_IT"
        assert babel.gettext("Welcome %(user)s!", user="Luca") == "Welcome Luca!"

File: RepoTransBench/target_projects/Python/Rust/python-babel_flask-babel/public_tests/test_public_multiple_apps.rs
use flask_babel_rs::*;
use serial_test::serial;

#[test]
#[serial]
fn test_public_multiple_apps() {
    let mut b = Babel::new();

    let mut app1 = DummyApp::new();
    b.init_app(&mut app1, Some("fr_FR"));

    let mut app2 = DummyApp::new();
    b.init_app(&mut app2, Some("it_IT"));

    {
        let _locale_guard = b.force_locale("fr_FR");
        assert_eq!(get_locale(), "fr_FR");
        let mut map = std::collections::HashMap::new();
        map.insert("user", "Marie");
        assert_eq!(b.gettext("Welcome %(user)s!", Some(map)), "Welcome Marie!");
    }

    {
        let _locale_guard = b.force_locale("it_IT");
        assert_eq!(get_locale(), "it_IT");
        let mut map = std::collections::HashMap::new();
        map.insert("user", "Luca");
        assert_eq!(b.gettext("Welcome %(user)s!", Some(map)), "Welcome Luca!");
    }
}